### PR TITLE
[SCAL-215269] added sample custom leaflet chart

### PR DESCRIPTION
**Pull request by UtkarshRai12**

### CODE REVIEW

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  Query,
+} from '@thoughtspot/ts-chart-sdk';
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import _ from 'lodash';
+import USjsonData from './ADMIN_DIV_2-US.json'
+import L from 'leaflet';
+
+function getDataModel(chartModel: ChartModel) {
+  const chartValue=chartModel.data?.[0].data.dataValue;
+  const updatedNameCounties= chartValue?.map(([name, value]) => {
```
Comment (harshmeetTS) [inline example/custom-leaflet-chart/src/geo-charts.ts:17]: use a linter

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  });
+  console.log(updatedNameCounties)
+  const USCountyData = USjsonData;
+  console.log(USCountyData)
+  const updatedCounties = updatedNameCounties?.map((county: any[]) => county[0]);
+  console.log(updatedCounties)
+  const filteredUSCountyData = USCountyData.features.filter((feature: any) => {
+    const idx=updatedCounties?.indexOf(feature.properties.NAME.toLowerCase());
+    if(idx==-1) return false
+    if(idx!==undefined) feature.properties.XData= updatedNameCounties?.[idx][1]??0;
```
Comment (harshmeetTS) [inline example/custom-leaflet-chart/src/geo-charts.ts:33]: remove unwanted console logs

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                      } as Query,
+                  ),
+          );
+          return queries;
+      },
+      renderChart: (ctx) => renderChart(ctx),
+  });
+
+  renderChart(ctx);
+})();
```
Comment (harshmeetTS) [inline example/custom-leaflet-chart/src/geo-charts.ts:148]: remove this EOL, "USE LINTER"

File: example/custom-leaflet-chart/styles.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+html {
+  height: 100%;
+}
+
+body,
+html,
+#map {
+  width: 105%;
+  height: 110%;
+}
```
Comment (harshmeetTS) [inline example/custom-leaflet-chart/styles.css:18]: why this type of styling is added ?

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    const USCountyData = USjsonData;
+    const updatedCounties = updatedNameCounties?.map(
+        (county: any[]) => county[0],
+    );
+    const filteredUSCountyDataWithNull = USCountyData.features.map(
+        (feature: any) => {
+            const idx = updatedCounties?.indexOf(
+                feature.properties.NAME.toLowerCase(),
+            );
+            if (idx === -1) return null;
```
Comment (yashvardhan-n) [inline example/custom-leaflet-chart/src/geo-charts.ts:33]: update if condition to idx ==-1 || idx !=undefined

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        minZoom: 1,
+        maxZoom: 18,
+    };
+    const geojsonFeature = {
+        type: 'FeatureCollection',
+        features: dataModel,
+    };
+    // magnification with which the map will start
+    const zoom = 3;
+    // // co-ordinates
```
Comment (yashvardhan-n) [inline example/custom-leaflet-chart/src/geo-charts.ts:66]: update comment

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        style: {
+            fillColor: '#000000',
+            color: '#f20b0b',
+            weight: 1,
+            opacity: 1,
+            fillOpacity: 0.0,
+        },
+        onEachFeature(feature, layer) {
+            layer.bindPopup(
+                `<h1>${feature.properties.NAME}</h1><p>Total Poverty Percentage: ${feature.properties.XData}</p>`,
```
Comment (yashvardhan-n) [inline example/custom-leaflet-chart/src/geo-charts.ts:84]: can we avoid hardcoding 'Total Poverty Percentage' here ?

File: example/custom-leaflet-chart/src/geo-charts.ts (hunk 5 above)
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/src/geo-charts.ts:33]: ack

File: example/custom-leaflet-chart/src/geo-charts.ts (hunk 6 above)
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/src/geo-charts.ts:66]: ack

File: example/custom-leaflet-chart/styles.css (hunk 4 above)
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/styles.css:18]: ack

**Review by yashvardhan-n — APPROVED** [review]
lgtm

File: example/custom-leaflet-chart/index.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>TS custom chart</title>
+    <link rel="stylesheet" href="styles.css" />
+    <link rel="stylesheet" href="./leaflet/leaflet.css" />
+  </head>
+  <body>
+    <div style="width:99vw; height:95vh; position: relative;">
+      <!-- <canvas id="chart" style="display:flex;"></canvas> -->
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/index.html]: ack

File: example/custom-leaflet-chart/package.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "custom-leaflet-chart",
+  "private": true,
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/package.json]: ack

File: example/custom-leaflet-chart/src/geo-charts.ts [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  console.log(filteredUSCountyData)
+  let config = {
+    minZoom: 1,
+    maxZoom: 18,
+  };
+  const geojsonFeature={
+    "type": "FeatureCollection",
+    "features": filteredUSCountyData
+  }
+  console.log(geojsonFeature);
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/src/geo-charts.ts]: ack

File: example/custom-leaflet-chart/src/geo-charts.ts (hunk 2 above)
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/src/geo-charts.ts]: ack

File: example/custom-leaflet-chart/src/geo-charts.ts (hunk 3 above)
Comment (UtkarshRai12) [reply example/custom-leaflet-chart/src/geo-charts.ts]: ack

**Review by yashvardhan-n — APPROVED** [review]
lgtm

**Pull request merged**